### PR TITLE
Fix a smattering of project warnings

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: common/src/service/ExceptionTrace.cs [hunk 1]
```diff
@@ -22,24 +25,24 @@ public ExceptionTrace(string eventSourceName)
             _eventSourceName = eventSourceName;
         }
 
-        public Exception AsError(Exception exception, EventTraceActivity activity = null)
```
Comment (drwill-ms) [inline common/src/service/ExceptionTrace.cs:25]: Unused parameters

File: common/src/service/ExceptionTrace.cs [hunk 2]
```diff
@@ -95,24 +99,16 @@ public void TraceHandled(Exception exception, string catchLocation, EventTraceAc
                 exception.ToStringSlim()));
 #endif
 
-            ////MessagingClientEtwProvider.Provider.HandledExceptionWithFunctionName(
```
Comment (drwill-ms) [inline common/src/service/ExceptionTrace.cs:98]: Several cases of removing dead code

File: common/src/service/ExceptionTrace.cs [hunk 3]
```diff
@@ -95,24 +99,16 @@ public void TraceHandled(Exception exception, string catchLocation, EventTraceAc
...
-            ////    activity, catchLocation, exception.ToStringSlim(), string.Empty);
-
             BreakOnException(exception);
         }
 
-        public void TraceUnhandled(Exception exception)
-        {
-            ////MessagingClientEtwProvider.Provider.EventWriteUnhandledException(this.eventSourceName + ": " + exception.ToStringSlim());
```
Comment (drwill-ms) [inline common/src/service/ExceptionTrace.cs:106]: Removing empty methods and call chain to them

File: common/src/service/Fx.cs [hunk 4]
```diff
@@ -153,10 +145,13 @@ public static Exception AssertAndFailFastService(string description)
...
                     // Throwing FatalException since our service does not morph/eat up fatal exceptions
                     // We should find the tracking bug in Azure for this issue, and remove the workaround when fixed by Azure
-                    Thread failFastWorkaroundThread = new Thread(delegate ()
-                    {
-                        throw new FatalException(failFastMessage);
-                    });
+                    var failFastWorkaroundThread = new Thread(
+                        delegate ()
+                        {
+#pragma warning disable CA2219 // Do not raise exceptions in finally clauses
```
Comment (drwill-ms) [inline common/src/service/Fx.cs:151]: I'm not sure wat the right fix is for this. Inside a finally you could be running code just before an exception gets thrown, so it makes sense.

I don't know what most of the Fx.cs class is for. Much of the code looks highly dubious.

**Review by vinagesh — APPROVED** [review]

**Pull request merged**